apps/rig/lib/rig/connection/codec.ex

```diff
@@ -1,29 +1,28 @@
 defmodule Rig.Connection.Codec do
   @moduledoc """
   Encode and decode a connection token, e.g., for correlation.
-
-  TODO: sign the token - using binary_to_term without signature verification is a threat.
   """
 
   @doc "Turn a pid into an url-encoded string."
   @spec serialize(pid) :: binary
   def serialize(pid) do
+    key = Confex.fetch_env!(:rig, Rig.Connection.Codec)[:secret_key]
```

Comment:
I think I'd prefer `use Rig.Config, [:secret_key]` instead

Reply:
I've applied the change you suggested, but instead of `:secret_key` I've used name `:codec_secret_key`, because `secret_key` is used elsewhere as alias for `jwt_secret_key` as far as I understand. I'm using `NODE_COOKIE` env var.

Reply:
well `:secret_key` in that context refers to the key used in the configuration _for this module_, which means there can't be a name clash here. That said, `:codec_secret_key` is fine, too.